core/api/transport/impl/ws/ws_listener_impl.hpp

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_CORE_API_TRANSPORT_IMPL_WS_LISTENER_IMPL_HPP
+#define KAGOME_CORE_API_TRANSPORT_IMPL_WS_LISTENER_IMPL_HPP
+
+#include "api/transport/listener.hpp"
+
+#include "api/transport/impl/ws/ws_session.hpp"
+#include "common/logger.hpp"
+
+namespace kagome::api {
+  /**
+   * @brief server which listens for incoming connection,
+   * accepts connections making session from socket
+   */
+  class WsListenerImpl : public Listener,
+                         public std::enable_shared_from_this<WsListenerImpl> {
+    enum class State { READY, WORKING, STOPPED = READY };
+    using Context = boost::asio::io_context;
+    using Acceptor = boost::asio::ip::tcp::acceptor;
+    using Endpoint = boost::asio::ip::tcp::endpoint;
+    using Duration = boost::asio::steady_timer::duration;
+    using Logger = common::Logger;
+
+   public:
+    using Session = WsSession;
+
+    /***
+     * Listener configuration
+     */
+    struct Configuration {
+      Endpoint endpoint{};  ///< listener endpoint
+    };
+
+    /**
+     * @param context reference to boost::asio::io_context instance
+     * @param endpoint loopback ip address to listen
+     * @param ws_config websocket session configuration
+     */
+    WsListenerImpl(Context &context,
+                   const Configuration &configuration,
+                   WsSession::Configuration ws_config);
```

Comment:
The same here, by the way. I just think that given that we actually have an interface named Session such typedefs might be low-key misleading.

Reply:
replaced by SessionImpl alias